styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesPodEventTranslator.java

```diff
@@ -131,101 +131,118 @@ public TerminationLogMessage(
+        generatedEvents.add(Event.terminate(workflowInstance, exitCode));
+        break;
+
+      default:
+        // no event
+        break;
+    }
+  }
+
+  private static void handleStarted(final WorkflowInstance workflowInstance, final RunState state,
+                                    final List<Event> generatedEvents) {
+    switch (state.state()) {
+      case PREPARE:
+      case SUBMITTED:
+        generatedEvents.add(Event.started(workflowInstance));
+        break;
+
+      default:
+        // no event
+        break;
+    }
+  }
+
+  private static void handleError(RunState state, List<Event> generatedEvents, Event event) {
+    switch (state.state()) {
+      case PREPARE:
+      case SUBMITTED:
+      case RUNNING:
+        generatedEvents.add(event);
+        break;
+
+      default:
+        // no event
+        break;
+    }
+  }
+
+  private static boolean isExited(Pod pod, Optional<ContainerStatus> mainContainerStatusOpt) {
+    switch (pod.getStatus().getPhase()) {
       case "Running":
         // Check if the main container has exited
-        exited = containerStatus.map(ContainerStatus::getState)
-                                .map(ContainerState::getTerminated)
-                                .isPresent();
-        if (exited) {
-          break;
+        final boolean mainContainerTerminated = mainContainerStatusOpt.map(ContainerStatus::getState)
+            .map(ContainerState::getTerminated)
+            .isPresent();
+        if (mainContainerTerminated) {
+          return true;
         }
 
-        // check that the main container is ready
-        // TODO: is checking for "ready" meaningful without a readiness probe configured?
-        started = containerStatus.map(ContainerStatus::getReady)
-                                 .orElse(false);
         break;
 
       case "Succeeded":
       case "Failed":
-        exited = true;
-        break;
+        return true;
 
       default:
         // do nothing
         break;
     }
 
-    if (started) {
-      switch (state.state()) {
-        case PREPARE:
-        case SUBMITTED:
-          generatedEvents.add(Event.started(workflowInstance));
-          break;
-
-        default:
-          // no event
-          break;
-      }
-    }
+    return false;
+  }
 
-    if (exited) {
-      switch (state.state()) {
-        case PREPARE:
-        case SUBMITTED:
-          generatedEvents.add(Event.started(workflowInstance));
-          // intentional fall-through
-
-        case RUNNING:
-          final Optional<Integer> exitCode = containerStatus.flatMap(cs ->
-              getExitCodeIfValid(workflowInstance.toKey(), pod, cs, stats));
-          generatedEvents.add(Event.terminate(workflowInstance, exitCode));
-          break;
-
-        default:
-          // no event
-          break;
-      }
+  private static boolean isStarted(Pod pod, Optional<ContainerStatus> mainContainerStatusOpt) {
```

Comment:
This method can be reduced to
```
return "Running".equals(pod.getStatus().getPhase()) && 
    mainContainerStatusOpt.map(ContainerStatus::getReady).orElse(false)
```

Reply:
Yeah. I thought that expression is a little long for eyes. But it seems ok.